### PR TITLE
[DYN-3512] Dynamo Menu's UX Update

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Update the Dynamo app menus to the new design

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### FYIs

Alfredo Pozo (@alfredo-pozo )
Roberto Tellez (@RobertGlobant20 )

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -569,6 +569,15 @@ public string DefaultPythonEngine
             }
         }
 
+        /// <summary>
+        /// Sets the visibility of the preferences menu option inside the Dynamo menu
+        /// The option will be disabled if in debug mode
+        /// </summary>
+        public bool IsVisiblePreferences
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:576]: Can we use converter https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/UI/Converters.cs? Any new property here may end up being APIs we do not intend to expose...

**Comment by mjkkirschner** [comment]
could we also add a screenshot of how this looks to the PR for reference?

**Comment by QilongTang** [comment]
> could we also add a screenshot of how this looks to the PR for reference?

Yes, requested on standup

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 1 above)
Comment (Astul-Betizagasti) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:576]: Sure, will look into it

Comment (Astul-Betizagasti) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Updated! Thanks for the comment, I didn't know about that converters.cs file before

**Comment by QilongTang** [comment]
Hi @Astul-Betizagasti Did you talk to @RobertGlobant20 about https://github.com/DynamoDS/Dynamo/pull/11517 yet? I think that is what I mean to hide over a specific debug mode.

**Comment by QilongTang** [comment]
Thanks @Astul-Betizagasti , since all the images icons are removed, can you also investigate if there is way to make the menu item look as the design (with no space before each item).

This PR LGTM

**Review by QilongTang — APPROVED** [review]

**Pull request merged**